eth_utils/currency.py

```diff
@@ -86,20 +87,21 @@ def to_wei(number, unit):
         raise TypeError("Unsupported type.  Must be one of integer, float, or string")
 
     s_number = str(number)
+    unit_value = units[unit.lower()]
 
     if d_number == 0:
         return 0
 
-    unit_value = units[unit.lower()]
-
     if d_number < 1 and '.' in s_number:
         with localcontext() as ctx:
             multiplier = len(s_number) - s_number.index('.') - 1
             ctx.prec = multiplier
             d_number = decimal.Decimal(value=number, context=ctx) * 10**multiplier
         unit_value /= 10**multiplier
 
-    result_value = d_number * unit_value
+    with localcontext() as ctx:
```

Comment:
I'm curious if this one is needed.  Do tests pass if it's removed?  The `d_number` already has the appropriate precision from the context above.

Reply:
The two `conversion_round_trip_*` tests fail without this. Specifically when the `number` is `'10000000000.000000000000000001'` it drops the trailing 1.